src/lightning/app/runners/cloud.py

```diff
@@ -521,6 +524,14 @@ def _resolve_existing_run_instance(
                     break
         return existing_cloudspace, existing_run_instance
 
+    def _resolve_existing_run_instances(self, project_id: str, name: str) -> List[Externalv1LightningappInstance]:
+        """Get all existing instances in the given project with the given name."""
+        run_instances = self.backend.client.lightningapp_instance_service_list_lightningapp_instances(
```

Comment:
What a horrible name, which does not tell me anything about what it does :D

Reply:
Changed to `_resolve_run_instances_by_name`